src/rust/rust-proto-new/src/protocol/service_client.rs

```diff
@@ -0,0 +1,5 @@
+/// Equivalent to the NamedService trait in tonic for server constructs.
+/// Pass this NAME to e.g. a healthcheck client.
+pub trait AssociatedNamedService {
```

Comment:
Could we instead call this `ServiceName`? `AssociatedNamedService` is a mouthful, and I'm confused about the meaning of the `Associated` part.